### PR TITLE
Completed MapReduce Functionality for Average Scores by Domain

### PR DESCRIPTION
**Pull request by jdfree**
I believe I've implemented what was requested here. There was no guideline as to how to handle invalid data in the input file, so I assumed that it should be ignored pending a requirement. I tried to keep everything simple and straightforward.

Ready for review.

### CODE REVIEW

File: interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java [hunk 1]
```diff
@@ -47,22 +51,49 @@ public static void main(String[] args) throws Exception {
...
-            // TODO: Implemnet the reducer phase
+            float sum = 0f;
+            int count = 0;
+
+            for (FloatWritable value : values) {
+                sum += value.get();
+                count++;
+            }
+
+            average.set(count > 0 ? sum / count : 0f);
```
Comment (jdfree) [inline interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java:106]: I don't think `reduce` would be called if there were zero values, but divide-by-zero is a nasty condition, so I check for it anyway.

File: interview-mr/pom.xml [hunk 2]
```diff
@@ -48,10 +48,5 @@
             <version>2.6.0</version>
             <scope>provided</scope>
         </dependency>
-        <dependency>
-            <groupId>com.google.guava</groupId>
```
Comment (jdfree) [inline interview-mr/pom.xml:52]: I didn't use guava, so no need to explicitly include it.

File: interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java [hunk 3]
```diff
@@ -31,38 +33,78 @@
...
  */
 public class AverageScore {
+
     public static void main(String[] args) throws Exception {
+        if(args.length < 2)
+            System.exit(2);
+        System.exit(mapReduce(args[0], args[1]));
+    }
+
+    public static int mapReduce(String inputPath, String outputPath) throws Exception {
```
Comment (jdfree) [inline interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java:43]: Separated this out for modularity.

File: interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java [hunk 4]
```diff
@@ -31,38 +33,78 @@
...
+                if(domain != null) {
+                    domainText.set(domain);
+                    context.write(domainText, number);
+                }
+
+            } catch (URISyntaxException e) {
+                // Invalid data - not a clean URL. No error reporting requirement, so ignore for now.
+            } catch (NumberFormatException e) {
+                // Invalid data - value wasn't a number (as in the column header line). 
+            	// No error reporting requirement, so ignore for now.
```
Comment (jdfree) [inline interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java:84]: Didn't think designing an error-reporting component was a reasonably-sized addition to a test that wasn't supposed to be taking very long.

File: interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+			
+			// Verify basic output format and number of entries
+			assertEquals(189, getOutput().size());
+		} catch (Exception e) {
+			fail("Exceptions should not be thrown, but threw "+e);
+		}
+	}
+	
+	@Test
+	public void mapReduceSpecificTest() {
```
Comment (jdfree) [inline interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java:70]: Kind of an 80/20 effort here. Tests all the basic edge cases, but doesn't get into many sequences/combinations of possible errors. Further testing would see diminishing returns.

A reasonable next addition would be a very large scale average computation, but that's a big piece of work to bite off.

File: interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+public class AverageScoreTest {
+	private final String outputPath = "junit-output";
+	AverageScore score;
+	
+	@Before
+	public void clean() {
+		try {
+			FileUtils.deleteDirectory(new File(outputPath));
+		} catch (IOException e) {
+			// ignore
```
Comment (jdfree) [inline interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java:29]: If we haven't run these tests before, there's nothing to delete.

File: interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Scanner;
+
+import com.sovrn.interview.mr.AverageScore;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AverageScoreTest {
+	private final String outputPath = "junit-output";
```
Comment (jdfree) [inline interview-mr/src/test/java/com/sovrn/interview/mr/AverageScoreTest.java:21]: Minor quibble, but could have made a separate output path for each test so all output is preserved after test failures, but it's pretty easy to run tests individually if they fail, too. I'm running out of lunch break :)
